packages/components/src/popover/index.js

```diff
@@ -113,16 +117,42 @@ class Popover extends Component {
 		this.rafHandle = window.requestAnimationFrame( this.refresh );
 	}
 
+	/**
+	 * Calling refreshOnAnchorMove
+	 * will only refresh the popover position if the anchor moves.
+	 */
+	refreshOnAnchorMove() {
+		const { getAnchorRect = this.getAnchorRect } = this.props;
+		const anchorRect = getAnchorRect( this.anchorNode.current );
+		const didAnchorRectChange = ! isShallowEqual( anchorRect, this.anchorRect );
+		if ( didAnchorRectChange ) {
+			this.anchorRect = anchorRect;
+			this.computePopoverPosition( this.state.popoverSize, anchorRect );
+		}
+	}
+
 	/**
 	 * Calling `refresh()` will force the Popover to recalculate its size and
 	 * position. This is useful when a DOM change causes the anchor node to change
 	 * position.
-	 *
-	 * @return {void}
 	 */
 	refresh() {
-		const popoverSize = this.updatePopoverSize();
-		this.computePopoverPosition( popoverSize );
+		const { getAnchorRect = this.getAnchorRect } = this.props;
+		const anchorRect = getAnchorRect( this.anchorNode.current );
+		const contentRect = this.contentNode.current.getBoundingClientRect();
+		const popoverSize = {
+			width: contentRect.width,
+			height: contentRect.height,
+		};
+		const didPopoverSizeChange = ! this.state.popoverSize || (
+			popoverSize.width !== this.state.popoverSize.width ||
+			popoverSize.height !== this.state.popoverSize.height
+		);
+		if ( didPopoverSizeChange ) {
+			this.setState( { popoverSize } );
+		}
+		this.anchorRect = anchorRect;
```

Comment:
That was the first thing I tried but it's not enough because the anchor don't move when we resize the window and we do need to update the popover in this case.